service/history/mutableStateBuilder.go

```diff
@@ -230,6 +230,12 @@ func (e *mutableStateBuilder) FlushBufferedEvents() error {
 	return nil
 }
 
+func (e *mutableStateBuilder) ApplyReplicationStateUpdates(failoverVersion int64) {
+	e.replicationState.CurrentVersion = failoverVersion
+	e.replicationState.LastWriteVersion = failoverVersion
+	e.replicationState.LastWriteEventID = e.hBuilder.nextEventID - 1
```

Comment:
can we make this attr NextEventID? so no need to do the -1 operation

Reply:
I'll put a TODO comment for this now.  Replication protocol currently uses LastWriteEventID so I don't want to mess with the terminology.  But I see your point.  I'll rename this when I start work on the replication protocol part of changes.